corehq/apps/cloudcare/static/cloudcare/js/formplayer/app.js

```diff
@@ -119,14 +119,7 @@ FormplayerFrontend.on('startForm', function (data) {
             showSuccess(gettext("Form successfully saved"), $("#cloudcare-notifications"), 10000);
 
             if(resp.nextScreen !== null && resp.nextScreen !== undefined) {
-                if (resp.nextScreen.hasOwnProperty('tree')) {
-                    // If the response has "tree" property then we know it's a form link
-                    // so we should start the form.
-                    FormplayerFrontend.trigger('startForm', resp.nextScreen);
-                } else {
-                    FormplayerFrontend.trigger("renderResponse", resp.nextScreen);
-                }
-
```

Comment:
we get this for free since we now use the `parse`'s routing which handles form navigation: https://github.com/dimagi/commcare-hq/blob/br/fix-undefined-app-id/corehq/apps/cloudcare/static/cloudcare/js/formplayer/entities/menu.js#L39